sql/hive/compatibility/src/test/scala/org/apache/spark/sql/hive/execution/HiveCompatibilitySuite.scala

```diff
@@ -221,9 +221,6 @@ class HiveCompatibilitySuite extends HiveQueryFileTest with BeforeAndAfter {
     "udf_when",
     "udf_case",
 
-    // Needs constant object inspectors
-    "udf_round",
```

Comment:
I notice `udf_round` appears in both `blacklist` and `whitelist`, and this test could be enabled if we fix the `NaN, null and Infinity` as I made in #6835.